src/core/dbus-execute.c

```
@@ -1001,6 +1001,63 @@ static int property_get_exec_dir_symlink(
         return sd_bus_message_close_container(reply);
 }
 
+static int property_get_exec_quota(sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        QuotaLimit *q = ASSERT_PTR(userdata);
+
+        assert(bus);
+        assert(reply);
+
+        return sd_bus_message_append(reply, "(tub)", q->quota_absolute, q->quota_scale, q->quota_enforce);
+}
+
+static int property_get_usage_quota(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Unit *u = ASSERT_PTR(userdata);
+        ExecContext *c;
+        bool enforce = false, accounting = false;
+        uint64_t current_usage_bytes = 0, limit_bytes = 0;
+
+        assert(bus);
+        assert(reply);
+
+        c = unit_get_exec_context(u);
+        if (c) {
+                ExecDirectoryType dt;
+                if (streq(property, "StateDirectoryQuotaUsage"))
+                        dt = EXEC_DIRECTORY_STATE;
+                else if (streq(property, "CacheDirectoryQuotaUsage"))
+                        dt = EXEC_DIRECTORY_CACHE;
+                else if (streq(property, "LogsDirectoryQuotaUsage"))
+                        dt = EXEC_DIRECTORY_LOGS;
+                else
+                        assert_not_reached();
+
+                const QuotaLimit *q;
+                q = &c->directories[dt].exec_quota;
+                enforce = q->quota_enforce;
+                accounting = q->quota_accounting;
+
+                if (q->quota_enforce || q->quota_accounting)
+                        unit_get_exec_quota_stats(u, c, dt, &current_usage_bytes, &limit_bytes);
```

Missing error check.

---

we don't need to check for error here. If call fails, the bus method shouldn't fail and just pass the default values. The method already logs failures as well

---

No, please also propagate the failure to the dbus client.